extensions/platform-http/component/src/main/java/org/apache/camel/component/http/server/spi/PlatformHttpEngine.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.http.server.spi;
+
+/**
+ * An abstraction of an HTTP Server engine on which HTTP endpoints can be deployed.
+ */
+public interface PlatformHttpEngine {
+
+    /**
+     * Deploys the given {@link HttpEndpointSpec} on this {@link PlatformHttpEngine}
+     *
+     * @param spec the {@link HttpEndpointSpec} to deploy
+     */
+    void deploy(HttpEndpointSpec spec);
```

Comment:
This can be simplified to something like:

    Consumer createConsumer(PlatformHttpEndpoint, Processor)

Remember than other than deploying you need also to take care of the "undeploy" i.e. when the component is stopped/suspended. 

As side effect this would made `HttpEndpointSpec` not needed.

Reply:
I tried to provide a simple and hard to break interface for the implementors because I tend to mistrust camel model as fragile due to its mutability and overused inheritance. You are right it can be done like you say and it might feel more natural to camelists so I followed your wish.

`QuarkusPlatformHttpConsumer` now also takes care for the stop, suspend and resume.